cypress/integration/jsPromisesPracticeDOM.spec.js

```diff
@@ -15,73 +12,116 @@ const thirdResolvedMsg = 'Third promise was resolved';
+    });
 
-  it('should resolve second promise on the right click', () => {
-    cy.clickButton({ button: 2 });
-    cy.checkMessageDisplayed(secondResolvedMsg);
-  });
+    it('should not show reject message if it was already resolved', () => {
+      page.body().click();
+      cy.tick(3000);
 
-  it('should resolve second promise on the left click', () => {
-    cy.clickButton({ button: 0 });
-    cy.checkMessageDisplayed(secondResolvedMsg);
-  });
+      page.notification().should('not.include.text', firstRejectedMsg);
+    });
 
-  it('should resolve third promise', () => {
-    cy.clickButton({ button: 2 });
-    cy.clickButton({ button: 0 });
-    cy.checkMessageDisplayed(thirdResolvedMsg);
+    it('should be resolved after the left click on 2999 millisecond', () => {
+      cy.tick(2999);
+      page.body().click();
+
+      page.notification().should('include.text', firstResolvedMsg);
+    });
+
+    it('should not be rejected after 2999 milliseconds of inactivity', () => {
+      cy.tick(2999);
+
+      page.notification().should('not.include.text', firstRejectedMsg);
+    });
+
+    it('should not show resolve message if it was already rejected', () => {
+      cy.tick(3000);
+      page.notification().should('include.text', firstRejectedMsg);
+
+      page.body().click();
+      page.notification().should('not.include.text', firstResolvedMsg);
+    });
   });
 
-  it('should resolve all promises', () => {
-    cy.clickButton({ button: 2 });
-    cy.clickButton({ button: 0 });
-    cy.checkMessageDisplayed(firstResolvedMsg);
-    cy.checkMessageDisplayed(secondResolvedMsg);
-    cy.checkMessageDisplayed(thirdResolvedMsg);
+  describe('secondPromise', () => {
+    it('should be resolved after the left click', () => {
+      page.body().click();
+
+      page.notification().should('include.text', secondResolvedMsg);
+    });
+
+    it('should be resolved after the right click', () => {
+      page.body().rightclick();
+
+      page.notification().should('include.text', secondResolvedMsg);
+    });
+
+    it('should not be resolved on page render', () => {
+      page.notification().should('not.include.text', secondResolvedMsg);
+    });
+
+    it('should not be resolved without clicks after delay', () => {
+      cy.tick(100000);
+
+      page.notification().should('not.include.text', secondResolvedMsg);
+    });
   });
 
-  it('should reject 1st promise, resolve 2nd and 3rd promises', () => {
-    // NOTE: waiting for reject
-    // eslint-disable-next-line cypress/no-unnecessary-waiting
-    cy.wait(3000);
-    cy.clickButton({ button: 0 });
-    cy.clickButton({ button: 2 });
-    cy.checkMessageDisplayed(firstRejectedMsg);
-    cy.checkMessageDisplayed(secondResolvedMsg);
-    cy.checkMessageDisplayed(thirdResolvedMsg);
+  describe('thirdPromise', () => {
+    it('should be resolved after the left and right click', () => {
+      page.body().click();
+      page.body().rightclick();
```

Comment:
You should also check that it works if click happen in any order
or with a big delay